api/src/paths/project/{projectId}/survey/{surveyId}/observations/create.ts

```diff
@@ -0,0 +1,152 @@
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { SYSTEM_ROLE } from '../../../../../../constants/roles';
+import { getDBConnection } from '../../../../../../database/db';
+import { HTTP400 } from '../../../../../../errors/CustomError';
+import { PostBlockObservationObject } from '../../../../../../models/block-observation-create';
+import { blockObservationIdResponseObject } from '../../../../../../openapi/schemas/observation';
+import { postBlockObservationSQL } from '../../../../../../queries/observation/observation-create-queries';
+import { getLogger } from '../../../../../../utils/logger';
+import { logRequest } from '../../../../../../utils/path-utils';
+
+const defaultLog = getLogger('paths/project/{projectId}/survey/{surveyId}/observations/create');
+
+export const POST: Operation = [
+  logRequest('paths/project/{projectId}/survey/{surveyId}/observations/create', 'POST'),
+  createBlockObservation()
+];
+
+POST.apiDoc = {
+  description: 'Create a new block observation.',
+  tags: ['block observation'],
+  security: [
+    {
+      Bearer: [SYSTEM_ROLE.SYSTEM_ADMIN, SYSTEM_ROLE.PROJECT_ADMIN]
+    }
+  ],
+  requestBody: {
+    description: 'Block Observation post request object.',
+    content: {
+      'application/json': {
+        schema: {
+          title: 'Block Observation request object',
+          type: 'object',
+          properties: {
+            block_name: {
+              title: 'Block name of the observation',
+              type: 'string'
+            },
+            start_datetime: {
+              type: 'string',
+              description: 'ISO 8601 date string'
+            },
+            end_datetime: {
+              type: 'string',
+              description: 'ISO 8601 date string'
+            },
+            observation_count: {
+              type: 'number'
+            },
+            observation_data: {
+              title: 'Block observation json data',
+              type: 'object',
+              properties: {}
+            }
+          }
+        }
+      }
+    }
+  },
+  responses: {
+    200: {
+      description: 'Block Observation response object.',
+      content: {
+        'application/json': {
+          schema: {
+            ...(blockObservationIdResponseObject as object)
```

Comment:
this entire endpoint needs to change to accept the entity type and then handle the insertion of block observation if type is block - cannot be only for block but instead needs to be generic